### PR TITLE
Clarify how to set HOMEBREW_GITHUB_API_TOKEN

### PR DESCRIPTION
**Pull request by JBallin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

It's currently unclear how to fix the GitHub API Token, I've attempted to clarify the output message. Open to feedback!

Unfortunately I'm unable to figure out why the tests aren't passing, would appreciate some guidance.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
@JBallin how about using `Utils::Shell.set_variable_in_profile` here?

**Comment by JBallin** [comment]
Great idea @MikeMcQuaid!

File: Library/Homebrew/utils/github.rb [hunk 1]
```diff
@@ -123,16 +123,18 @@ def api_credentials_error_message(response_headers, needed_scopes)
             Your macOS keychain GitHub credentials do not have sufficient scope!
             Scopes they need: #{needed_human_scopes}
             Scopes they have: #{credentials_scopes}
-            Create a personal access token: #{ALL_SCOPES_URL}
-            and then set HOMEBREW_GITHUB_API_TOKEN as the authentication method instead.
+            Create a personal access token:
+             #{ALL_SCOPES_URL}
+            #{Utils::Shell.set_variable_in_profile("HOMEBREW_GITHUB_API_TOKEN", "YOUR_TOKEN")}
```
Comment (reitermarkus) [inline Library/Homebrew/utils/github.rb]: The indentation should be 2 spaces everywhere.

File: Library/Homebrew/cmd/gist-logs.rb [hunk 2]
```diff
@@ -43,7 +43,7 @@ def gistify_logs(f)
       puts <<~EOS
         You can create a new personal access token:
          #{GitHub::ALL_SCOPES_URL}
-        and then set the new HOMEBREW_GITHUB_API_TOKEN as the authentication method.
+        #{Utils::Shell.set_variable_in_profile("HOMEBREW_GITHUB_API_TOKEN", "YOUR_TOKEN")}
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/gist-logs.rb]: I think `$YOUR_TOKEN` would be more obvious, perhaps?

Comment (JBallin) [reply Library/Homebrew/cmd/gist-logs.rb]: I tried that initially but I'm having issues with `Utils::Shell.sh_quote` adding a backslash before the `$`. Any pointers?

result of passing `$YOUR_TOKEN`:
`'export HOMEBREW_GITHUB_API_TOKEN=\$YOUR_TOKEN'`

Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/gist-logs.rb]: In that case perhaps a lowercase `your_token_here` or similar would be better?

File: Library/Homebrew/utils/github.rb (hunk 1 above)
Comment (JBallin) [reply Library/Homebrew/utils/github.rb]: Fixed!

**Comment by MikeMcQuaid** [comment]
@JBallin `brew tests` seems to be failing.

**Comment by JBallin** [comment]
@MikeMcQuaid, the tests pass locally. There's an issue with test-bot that seems to be affecting other PR's as well.

`GitHub::search_issues` returns `Validation Failed`. My best guess is that it's an issue with hitting the GitHub API on Homebrew's end? Should I open an issue about this?

Here's the failure output from test-bot:
```
2018-10-15T04:19:31.2064790Z Failures:
2018-10-15T04:19:31.2065400Z 
2018-10-15T04:19:31.2066170Z   1) GitHub::search_issues queries GitHub issues with the passed parameters
2018-10-15T04:19:31.2067140Z      Failure/Error: results = subject.search_issues("brew search", repo: "Homebrew/brew", author: "avetamine", is: "closed")
2018-10-15T04:19:31.2067900Z 
2018-10-15T04:19:31.2068610Z      GitHub::Error:
2018-10-15T04:19:31.2069340Z        Validation Failed
2018-10-15T04:19:31.2071170Z      # ./utils/github.rb:237:in `raise_api_error'
2018-10-15T04:19:31.2071890Z      # ./utils/github.rb:194:in `open_api'
2018-10-15T04:19:31.2073140Z      # ./utils/github.rb:309:in `search'
2018-10-15T04:19:31.2074730Z      # ./utils/github.rb:242:in `search_issues'
2018-10-15T04:19:31.2076060Z      # ./test/utils/github_spec.rb:34:in `block (3 levels) in <top (required)>'
2018-10-15T04:19:31.2076720Z      # ./test/spec_helper.rb:134:in `block (2 levels) in <top (required)>'
2018-10-15T04:19:31.2077540Z      # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:123:in `block in run'
2018-10-15T04:19:31.2078190Z      # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:110:in `loop'
2018-10-15T04:19:31.2078840Z      # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:110:in `run'
2018-10-15T04:19:31.2079480Z      # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
2018-10-15T04:19:31.2080080Z      # ./test/spec_helper.rb:97:in `block (2 levels) in <top (required)>'
2018-10-15T04:19:31.2080690Z      # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:123:in `block in run'
2018-10-15T04:19:31.2081310Z      # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:110:in `loop'
2018-10-15T04:19:31.2081950Z      # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:110:in `run'
2018-10-15T04:19:31.2082580Z      # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
2018-10-15T04:19:31.2083250Z      # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:37:in `block (2 levels) in setup'
2018-10-15T04:19:31.2083890Z      # ./vendor/bundle/ruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block (2 levels) in <top (required)>'
2018-10-15T04:19:31.2084020Z 
2018-10-15T04:19:31.2084120Z Finished in 3 minutes 15.5 seconds (files took 2.51 seconds to load)
2018-10-15T04:19:31.2084230Z 507 examples, 1 failure
2018-10-15T04:19:31.2084300Z 
2018-10-15T04:19:31.2084380Z Failed examples:
2018-10-15T04:19:31.2084450Z 
2018-10-15T04:19:31.2084560Z rspec ./test/utils/github_spec.rb:33 # GitHub::search_issues queries GitHub issues with the passed parameters
```

**Comment by MikeMcQuaid** [comment]
Thanks again @JBallin!

**Pull request merged**